### ISSUE TITLE
iotedge debian purge leaves data in '/var/lib/iotedge'

### ISSUE THREAD

**Issue by gri6507** [Azure/iotedge#2306]
After installing iotedge debian, then manually configuring `/etc/iotedge/config.yaml`, followed by `systemctl restart iotedge`, and then waiting for the IoT Edge device to become fully functional, it looks like `iotedge` creates some content in '/var/lib/iotedge' directory. However, during `apt purge iotedge`, that content is not cleaned up, causing 

```bash
dpkg: warning: while removing iotedge, directory '/var/lib/iotedge' not empty so not removed 

gri6507@abyss:~>ls -al /var/lib/iotedge
total 16
drwxr-xr-x  4 iotedge iotedge 4096 Jan 13 15:51 .
drwxr-xr-x 38 root    root    4096 Jan 13 14:33 ..
drwxr-xr-x  2 iotedge iotedge 4096 Jan 13 15:51 cache
drwxr-xr-x  5 iotedge iotedge 4096 Jan 13 14:34 hsm
```

**Comment by arsing** [comment]
Yes, that is the home directory of the `iotedge` user, and contains working files required for the `iotedge` service at runtime.

The files must be kept across updates, but can be removed when the package is uninstalled or purged.

I see it's currently listed in [`debian/dirs`](https://github.com/Azure/iotedge/blob/b1e6ba543f5be7b4c6cb9b6d7d7f2d97b401e203/edgelet/contrib/debian/dirs) along with the logs directory. Perhaps they should be created in `postinst` and removed in `postrm` instead?

**Comment by gri6507** [comment]
> Perhaps they should be created in `postinst` and removed in `postrm` instead?

That would make sense.

**Comment by gri6507** [comment]
I created https://github.com/Azure/iotedge/pull/2350 for this

**Comment by arsing** [comment]
Fixed by #2350
